### ISSUE TITLE
Some minor improvements for better integration

### ISSUE THREAD

**Issue by otavio** [raphamorim/rio#94]
I am currently working on integrating Rio on NixOS and Home Manager, and it would greatly improve the integration if we could add support for some minor features. These include:
- [ ] support `--help`
- [ ] support  `-e` and `--command`
- [ ] support `--version`

**Comment by raphamorim** [comment]
Regarding `-e` and `--command`: would those params lead to the execution of a received _string_ in the initialization? (e.g: `rio -e "print 2"`)?

**Comment by otavio** [comment]
Yes; most terminal emulators support it. Doing so I can hook it inside NixOS automated tests 👍🏼

**Comment by raphamorim** [comment]
@otavio added all those three commands in https://github.com/raphamorim/rio/commit/fe44ac3a06cffa56c220688821c8e8eb1cad69b7

```
cargo build && ./target/debug/rio --help
   Compiling teletypewriter v2.0.0 (/Users/hugoamor/Documents/a/rio/teletypewriter)
   Compiling rio v0.0.61 (/Users/hugoamor/Documents/a/rio/rio)
    Finished dev [unoptimized + debuginfo] target(s) in 2.79s
Rio terminal app

Usage: rio [OPTIONS]

Options:
  -e, --command <COMMAND>...  Command and args to execute (must be last argument)
  -h, --help                  Print help
  -V, --version               Print version
```
  
```
  cargo build && ./target/debug/rio --version
    Finished dev [unoptimized + debuginfo] target(s) in 0.17s
rio 0.0.61
```

```
cargo build && ./target/debug/rio -e "print hey"
```

<img width="712" alt="Screenshot 2023-06-14 at 19 24 33" src="https://github.com/raphamorim/rio/assets/3630346/6baf4bb9-3640-48a0-b9b9-1097cff65a30">

**Comment by otavio** [comment]
The `-e` worked, but an improvement is if exits when the command ends.

**Comment by otavio** [comment]
Is it possible to run rio using software rendering?

See the test output:
```
vm-test-run-terminal-emulator-rio> machine # [    6.601198] sudo[1002]:     root : TTY=hvc0 ; PWD=/tmp ; USER=alice ; COMMAND=/run/current-system/sw/bin/run-in-this-term-wrapped report-success
vm-test-run-terminal-emulator-rio> machine # [    6.604942] sudo[1002]: pam_unix(sudo:session): session opened for user alice(uid=1000) by (uid=0)
vm-test-run-terminal-emulator-rio> machine # error: XDG_RUNTIME_DIR is invalid or not set in the environment.
vm-test-run-terminal-emulator-rio> machine # libEGL warning: DRI2: failed to authenticate
vm-test-run-terminal-emulator-rio> machine # error: XDG_RUNTIME_DIR is invalid or not set in the environment.
vm-test-run-terminal-emulator-rio> machine # error: XDG_RUNTIME_DIR is invalid or not set in the environment.
vm-test-run-terminal-emulator-rio> machine # [    7.027512] dhcpcd[710]: eth0: Router Advertisement from fe80::2
```

**Comment by raphamorim** [comment]
> The -e worked, but an improvement is if exits when the command ends.

Does not work `./target/debug/rio -e "print hey;exit;"`?

Or exiting after the execution of a single command (`print 1`) is a pattern from other terms? Then if yes, wonder how works. Does it closes straight way or add put the main process to sleep a bit before close.

**Comment by raphamorim** [comment]
> Is it possible to run rio using software rendering?
>
> See the test output:

No (unfortanelly) Rio doesn't provide a software render as fallback. In the case you described seems related to https://github.com/gfx-rs/wgpu/issues/2123. Are you using docker (what config)?

Maybe try with lower adapter as well (`perfomance=low`)

**Comment by otavio** [comment]
An example is:
https://github.com/alacritty/alacritty/blob/afffdbe612cb8b573016184400e0deeb0137ccb9/alacritty/src/cli.rs#L177-L180

It handles the command as said. It seems like the terminals also handle the `SHELL` environment variable.

**Comment by raphamorim** [comment]
hmm, shell by options right? Now only can use SHELL variables by config file. Also, think would be good allow a flag like `--config <custom-path>`

**Comment by raphamorim** [comment]
Thank you, is the automated tests blocked due to xdg runtime error? If yes, we can move that also to other issue or feel free to ping me on discord.
